### DISCUSSION TITLE
Is there a way to skip a test that has already started?

### DISCUSSION THREAD

**Discussion by biltongza** [cypress-io/cypress#9346] [Questions and Help]
We have a web form that continuously runs validation in the background when form controls change. At the moment, we don't have full control of the backend's state so when we run the same form inputs every day, they get picked up as duplicates and it breaks our test (timing out waiting for something to be visible).

I've managed to get to a point where I can detect the first validate call we run, inspect it for errors, and fail immediately by throwing a javascript error with a descriptive message.

This is a decent start, but I'd like to report these tests as skipped or pending instead of failed, separating them from the other failed tests. We know this scenario and would like to exclude it from "real" failed scenarios where something else has gone wrong that we aren't expecting.

Obviously the correct way to handle this scenario is to put the backend into a known state where our tests can run and reset the state between runs. It's something we are working towards but we are looking for an interim solution.

Is there any way to change how cypress reports these tests? Can I skip the test if it has already started?

**Comment by todd-m-kemp** [comment]
It sounds like making use of `this.skip()` would be helpful to you.

You can start the test as normal, at at the point in the test you've decided you no longer wish to proceed, use `this.skip`. The test will stop running at that point and be considered a skipped just as if you had marked the test with `it.skip`.

There's a description of how to use this here: http://mochajs.org/#inclusive-tests .

Hope this helps!

**Reply by unikitty37** [reply]
I'm confused — when I put `this.skip()` in a Cypress test, it aborts with `TypeError: Cannot read property 'skip' of undefined`. How do you actually use it in the context of a Cypress test?

**Reply by jennifer-shehane** [reply]
@unikitty37 Make sure you are using the function argument instead of the arrow function to preserve the correct context of `this`.

```js
it('test', () => {
  // ❗️ fails, wrong 'this' context
  this.skip()
})

it('test2', function () {
  // 👍 skips test as intended
  this.skip()
})
```

**Comment by bahmutov** [comment]
Take a look at https://github.com/cypress-io/cypress-skip-test
